pkg/cloudproduct/cloudproduct.go

```diff
@@ -70,7 +124,7 @@ func New(ctx context.Context, product string, kc *kubernetes.Clientset) (CloudPr
 	case genericProduct:
 		return generic.New()
 	}
-	return nil, fmt.Errorf("unknown cloud product: %q", product)
+	return nil, nil, fmt.Errorf("unknown cloud product: %q", product)
```

Comment:
I missed this the first time around, but please use `errors.Errorf()` so that we can capture a stacktrace at error time.

Reply:
Done